### PR TITLE
Marshmallow Based Event Schema Validation

### PR DESCRIPTION
**Pull request by piotrekno1**
Coversation of the discussion started in: https://github.com/lukaszb/cq/pull/21

I know that for now this looks super ugly but wanted to get your feedback on:

* Placement of the validation code. Validation is controlled by `Repository` class. 
* How do you like the Schema registration API? It mimics what is done for mutators.
* Do you like the fact that only event data is validated (static fields of an event are skipped to avoid repetetive work).

Things that needs to be changed if we agree on the points above:
* Decide where to place schema definitions in examples (`schema.py` ?).
* Add tests
* Relax the setup.py import
* Strict mode to enforce validation (?)

### CODE REVIEW

**Comment by lukaszb** [comment]
I really like it, plain and simple.

> - Placement of the validation code. Validation is controlled by Repository class.

:+1: This way we'd keep `Storage.store` raw (so it's good for example in testing). People should use `Repository` for real usage anyway.

> - How do you like the Schema registration API? It mimics what is done for mutators.

Looks cool. I wonder if we can combine two things, though - event definition and mutator. This, however, can be done in a next step.

> - Do you like the fact that only event data is validated (static fields of an event are skipped to avoid repetetive work).

Sure, we can extend that in future but I'm not sure if that would be needed at all. Data is all we should care for now.

**Comment by piotrekno1** [comment]
Please see the proposal.

File: cq/aggregates.py [hunk 1]
```diff
@@ -19,17 +20,39 @@ def mutate(self, event):
...
+    def validate(cls, event_name, event_data):
+        schema_cls = cls.get_schema(event_name)
+        if not schema_cls:
+            return True
+
+        schema = schema_cls()
+        result = schema.load(event_data)
+        if result.errors:
+            msg = "Error validatng %s.%s event. Details: %s." % (cls.get_name(), event_name, result.errors)
+            raise SchemaValidationError(msg)
```
Comment (lukaszb) [inline cq/aggregates.py]: please propagate errors in a raised exception instance

**Comment by piotrekno1** [comment]
@lukaszb Not sure I understood. Does https://github.com/lukaszb/cq/pull/40/commits/1d50e993a7263520533415b3b199f75c048b2150 address your feedback?

**Comment by lukaszb** [comment]
@piotrekno1 exactly, thanks :)

**Pull request merged**